lib/spack/spack/relocate.py

```diff
@@ -385,57 +386,79 @@ def macholib_get_paths(cur_path):
     return (rpaths, deps, ident)
 
 
-def modify_elf_object(path_name, new_rpaths):
-    """
-    Replace orig_rpath with new_rpath in RPATH of elf object path_name
-    """
+def _set_elf_rpaths(target, rpaths):
+    """Replace the original RPATH of the target with the paths passed
+    as arguments.
 
-    new_joined = ':'.join(new_rpaths)
+    This function uses ``patchelf`` to set RPATHs.
 
-    # if we're relocating patchelf itself, use it
-    bak_path = path_name + ".bak"
+    Args:
+        target: target executable. Must be an ELF object.
+        rpaths: paths to be set in the RPATH
 
-    if path_name[-13:] == "/bin/patchelf":
-        shutil.copy(path_name, bak_path)
-        patchelf = executable.Executable(bak_path)
-    else:
-        patchelf = executable.Executable(_patchelf())
+    Returns:
+        A string concatenating the stdout and stderr of the call
+        to ``patchelf``
+    """
+    # Join the paths using ':' as a separator
+    rpaths_str = ':'.join(rpaths)
+
+    # If we're relocating patchelf itself, make a copy and use it
+    bak_path = None
+    if target.endswith("/bin/patchelf"):
+        bak_path = target + ".bak"
+        shutil.copy(target, bak_path)
 
+    patchelf, output = executable.Executable(bak_path or _patchelf()), None
     try:
-        patchelf('--force-rpath', '--set-rpath', '%s' % new_joined,
-                 '%s' % path_name, output=str, error=str)
+        # TODO: revisit the use of --force-rpath as it might be conditional
+        # TODO: if we want to support setting RUNPATH from binary packages
+        patchelf_args = ['--force-rpath', '--set-rpath', rpaths_str, target]
```

Comment:
@gartung If I understand the code correctly this function will always set RPATHs and never set RUNPATHs regardless of what is in `config.yaml`? Is it correct?

Reply:
As far as I know patchelf only handles rpaths.

Reply:
I'll experiment with that. I think if we don't give the ``--force-rpath` option then the relocation will be done on RUNPATH.